# <div class="alert alert-block alert-info"> Sentimental Analysis of Donald Trump's tweets </div>

In [2]:
import re

In [3]:
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = "all"

In [4]:
def stripSpecialCharacters(word):
    return word.strip("-\(.!,?\) '\"/:_;\\")

<img src="animated-gif-download.gif"/>

### <div class="alert alert-block alert-warning"> Converting the list of words into dictionary to find unique words and save operational time and effort. </div>

In [5]:
tweets_file = open('Trump_Raw_Tweets_Fall22.txt','r')

dictionary={}
lines=[]
for line in tweets_file:
    lower_line=line.lower()
    no_url_line = re.sub(r'http\S+|\'b\'|"b\'|\'b"',r'', lower_line)
    no_unicode_line=re.sub(r'\\xe2\\x80\\x99',"'",no_url_line)
    no_unicode_line=re.sub(r'\\x[0-9a-f]{2}',r'',no_unicode_line)
    lines.extend(no_unicode_line.split('\\n'))
#word_list=[]
for line in lines:
    list1=line.strip().split()
    for w in list1:
        w1=stripSpecialCharacters(w)
        if len(w1)> 0:
            #word_list.append(w1)
            if w1 not in dictionary:
                dictionary[w1]=1
            else:
                dictionary[w1]=dictionary[w1]+1
total_words=0
for v in dictionary.values():
    total_words+=v
print('Total number of words after data cleaning: ', total_words)
# total=len(word_list)
# print('Total number of words in tweets file: ', total)

Total number of words after data cleaning:  56264


### <div class="alert alert-block alert-success"> Number of positive words: 3081 </div>

In [171]:
positive_file = open('/Users/vrushtikhajanchi/Desktop/MSBA/Fall\'22/Python/HW1/positive.txt','r')
positive_words=set()
for line in positive_file:
    positive_words.add(line.rstrip('\n'))
#print(positive)
unique_positive=0
total_positive=0
for i in positive_words:
    if len(i) > 0 and i in dictionary:
        unique_positive+=1
        total_positive+=dictionary[i]
print('Unique positive words:', unique_positive)
print('Total positive words: ', total_positive)

Unique positive words: 229
Total positive words:  3081


### <div class="alert alert-block alert-danger"> Number of negative words: 2358 </div>

In [172]:
negative_file = open('/Users/vrushtikhajanchi/Desktop/MSBA/Fall\'22/Python/HW1/negative.txt','r')
negative_words=set()
for line in negative_file:
    negative_words.add(line.rstrip('\n'))
#print(negative)
unique_negative=0
total_negative=0
for neg in negative_words:
    if len(neg) > 0 and neg in dictionary:
        unique_negative-=1
        total_negative-=dictionary[neg]
    
    
print('Unique negative words:',unique_negative)
print('Total negative words: ', total_negative)

Unique negative words: -483
Total negative words:  -2358


### <div class="alert alert-block alert-warning">Number of stop words: 27739</div>

In [178]:
stop_file = open('/Users/vrushtikhajanchi/Desktop/MSBA/Fall\'22/Python/HW1/stop_words.txt','r')
stop_words=set()
for line in stop_file:
    stop_words.add(line.rstrip('\n'))
#print(stop)
unique_stop=0
total_stop=0
for s in stop_words:
    if len(s) == 0 or s in negative_words or s in positive_words:
        continue
    if s in dictionary:
        unique_stop+=1
        total_stop+=dictionary[s]
print('Unique stop words:',unique_stop)
print('Total stop words: ', total_stop)
us1=0
ts1=0
for s in stop_words:
    if len(s) == 0:
        continue
    if s in dictionary:
        us1+=1
        ts1+=dictionary[s]
print(us1)
print(ts1)

Unique stop words: 564
Total stop words:  27739
586
28221


In [180]:
stop_file1 = open('/Users/vrushtikhajanchi/Desktop/MSBA/Fall\'22/Python/HW1/stop_words.txt','r')
stop_words1=set()
for line in stop_file1:
    stop_words1.add(line.rstrip('\n'))
#print(stop)
unique_stop1=0
total_stop1=0
for s in stop_words1:
    if len(s) == 0 :
        continue
    if s in dictionary:
        unique_stop1+=1
        total_stop1+=dictionary[s]
print('Unique stop words:',unique_stop1)
print('Total stop words: ', total_stop1)

positive_file1 = open('/Users/vrushtikhajanchi/Desktop/MSBA/Fall\'22/Python/HW1/positive.txt','r')
positive_words1=set()
for line in positive_file1:
    positive_words1.add(line.rstrip('\n'))
#print(positive)
unique_positive1=0
total_positive1=0
for i in positive_words1:
    if len(i) > 0 and i in dictionary and i not in stop_words1:
        unique_positive1+=1
        total_positive1+=dictionary[i]
print('Unique positive words:', unique_positive1)
print('Total positive words: ', total_positive1)

negative_file1 = open('/Users/vrushtikhajanchi/Desktop/MSBA/Fall\'22/Python/HW1/negative.txt','r')
negative_words1=set()
for line in negative_file1:
    negative_words1.add(line.rstrip('\n'))
#print(negative)
unique_negative1=0
total_negative1=0
for neg in negative_words1:
    if len(neg) > 0 and neg in dictionary and neg not in stop_words1:
        unique_negative1-=1
        total_negative1-=dictionary[neg]
    
    
print('Unique negative words:',unique_negative1)
print('Total negative words: ', total_negative1)

Unique stop words: 586
Total stop words:  28221
Unique positive words: 211
Total positive words:  2611
Unique negative words: -479
Total negative words:  -2346


### <div class="alert alert-block alert-warning">Number of other words: 23086</div>

In [174]:
other_words=set()
unique_others=0
total_others=0
for d in dictionary:
    if d not in positive_words and d not in negative_words and d not in stop_words:
        other_words.add(d)
        unique_others+=1
        total_others+=dictionary[d]
print('Unique other words: ',unique_others)
print('Total other words: ', total_others)
#print(others)

Unique other words:  4834
Total other words:  23086


In [175]:
total2 = total_positive+abs(total_negative)+total_stop+total_others
print(total2)

56264


## <div class="alert alert-block alert-info">Total number of words: 56264

**Positive words:** 3081 
    
**Negative words:** 2358 
    
**Stop words:** 27739
    
**Other words:** 23086 </div>

## <div class="alert alert-block alert-info"> Ratio:</div>

In [177]:
def ratio(a,b):
    r = (a/b)
    return round(r, 3)
#print(ratio(total_positive))    
print('Ratio of positive words to total words: ', ratio(total_positive,total))
print('Ratio of negative to total words: ', ratio(abs(total_negative), total))
print('Ratio of stop words to total words: ', ratio(total_stop, total))
print('Ratio of other words to total words: ', ratio(total_others, total))
print('Ratio of positive words to negative words: ', ratio(total_positive, abs(total_negative)))

Ratio of positive words to total words:  0.055
Ratio of negative to total words:  0.042
Ratio of stop words to total words:  0.493
Ratio of other words to total words:  0.41
Ratio of positive words to negative words:  1.307


## <div class="alert alert-block alert-info">Analysis: </div>

### <div class="alert alert-block alert-info"> The overall sentiment of tweets is <i> positive</i> because total number of positive words exceeds negative words and positive words are around 31% more than negative words. This is a strong inclination towards positive sentiments. </div>

In [164]:
# for o in other_words:
#     if o in negative_words:
#         print(o)
#     elif o in positive_words:
#         print(o)
#     elif o in stop_words:
#         print(o)

In [103]:
#no_unicode_line=re.sub(r'[\\x00-\\x7f]','',no_url_line)
#     no_unicode_line = no_unicode_line.replace('\\xf0\\x9f\\x87\\xba','').replace('\\xf0\\x9f\\x87\\xb8','').replace('\\xf0\\x9f\\x8f\\x9b','').replace('\\xf0\\x9f\\x98\\x86','').replace('\\xf0\\x9f\\xa4\\xa3','').replace('\\xf0\\x9f\\xa7\\xb5','').replace('\\xf0\\x9f\\xa4\\x94','').replace('\\xf0\\x9f\\x98\\x82','').replace('\\xf0\\x9f\\x92\\xaf','').replace('\\xf0\\x9f\\x94\\xa5','').replace('\\xf0\\x9f\\x8e\\xa4','').replace('\\xe2\\xac\\x87','').replace('\\xef\\xb8\\x8f','').replace('\\xf0\\x9f\\x98\\xa1','').replace('\\xf0\\x9f\\x99\\x84','').replace('\\xc2\\xbb','').replace('\\xe2\\x98\\x84','').replace('\\xef\\xb8\\x8f','').replace('\\xf0\\x9f\\xa5\\xba','').replace('\\xf0\\x9f\\x8c\\x8a','').replace('\\xf0\\x9f\\x8c\\x8a','').replace('\\xe2\\x9c\\x8c','').replace('\\xf0\\x9f\\x8f\\xbf','').replace('\\xe2\\x9a\\xa1','').replace('\\xef\\xb8\\x8f','').replace('\\xf0\\x9f\\x98\\x82','').replace('\\xf0\\x9f\\xa7\\xb5','').replace('\\xf0\\x9f\\x91\\x8d','').replace('\\xf0\\x9f\\x8f\\xbb','').replace('\\xf0\\x9f\\x8f\\xbc','').replace('\\xf0\\x9f\\xa4\\xac','').replace('\\xf0\\x9f\\x98\\xb3','').replace('\\xf0\\x9f\\x98\\x80','')
#     no_unicode_line = no_unicode_line.replace('\\xf0\\x9f\\x93\\x96','').replace('\\xf0\\x9f\\xa4\\xb7','').replace('\\xf0\\x9f\\x8f\\xbd','').replace('\\xe2\\x80\\x8d\\','').replace('xe2\\x99\\x80','').replace('\\xef\\xb8\\x8f','').replace('\\xf0\\x9f\\x99\\x8f','').replace('\\xf0\\x9f\\x8f\\xbb','').replace('\\xf0\\x9f\\xa4\\xa1','').replace('\\xf0\\x9f\\x99\\x8f','').replace('\\\\xf0\\x9f\\x98\\x8a','').replace('\\xf0\\x9f\\x98\\x82','').replace('\\xf0\\x9f\\x92\\xaf','').replace('\\xf0\\x9f\\x94\\xa5','').replace('\\xf0\\x9f\\x8e\\xa4','').replace('\\xe2\\xac\\x87\\xef\\xb8','').replace('\\x8f\\xf0\\x9f\\x98\\xa1','').replace('\\xf0\\x9f\\x99\\x84','').replace('\\xc2\\xbb','').replace('\\xe2\\x98\\x84','').replace('\\xef\\xb8\\x8f','').replace('\\xf0\\x9f\\xa5\\xba','').replace('\\xf0\\x9f\\x8c\\x8a','').replace('\\xf0\\x9f\\x8c\\x8a','').replace('\\xf0\\x9f\\x8f\\xbb','').replace('\\xf0\\x9f\\x8f\\xbb','').replace('\\xf0\\x9f\\x94\\xa5','').replace('\\xf0\\x9f\\x98\\x85','').replace('\\xe2\\xac\\x87','').replace('\\xef\\xb8\\x8f','').replace('\\xf0\\x9f\\x92\\x95','').replace('\\xf0\\x9f\\x91\\x80','').replace('\\xe2\\x9c\\x85','').replace('\\xc2\\xb0','').replace('\\xf0\\x9f\\xa4\\xa6','').replace('\\xf0\\x9f\\x8f\\xbb','').replace('\\xe2\\x80\\x8d','').replace('\\xe2\\x99\\x80','').replace('\\xef\\xb8\\x8f','')
#     no_unicode_line = no_unicode_line.replace('\\xf0\\x9f\\x8c\\xb9','').replace('\\xe3\\x80\\x80','').replace('\\xe2\\x99\\xa5','').replace('\\xef\\xb8\\x8f','').replace('\\xe2\\x99\\xa5','').replace('\\xef\\xb8\\x8f','').replace('\\xe2\\x99\\xa5','').replace('\\xef\\xb8\\x8f','').replace('\\xf0\\x9f\\x8c\\xb9','').replace('\\xf0\\x9f\\xa5\\x82','').replace('\\xe3\\x80\\x80','').replace('\\xf0\\x9f\\xa4\\xb7','').replace('\\xe2\\x80\\x8d','').replace('\\xe2\\x99\\x80','').replace('\\xef\\xb8\\x8f','').replace('\\xe2\\x9d\\xa4','').replace('\\xef\\xb8\\x8f','').replace('\\xf0\\x9f\\x87\\xb1','').replace('\\xf0\\x9f\\x87\\xb7','').replace('\\xe2\\x9d\\xa4','').replace('\\xef\\xb8\\x8f','').replace('\\xf0\\x9f\\x99\\x81','').replace('\\xf0\\x9f\\xa4\\xa5','').replace('\\xf0\\x9f\\xa4\\xa5','').replace('\\xf0\\x9f\\x87\\xb7','').replace('\\xf0\\x9f\\x87\\xba','').replace('\\xf0\\x9f\\x87\\xb7','').replace('\\xf0\\x9f\\x87\\xba','').replace('\\xf0\\x9f\\x91\\xb9','').replace('\\xf0\\x9f\\xa4\\xad','').replace('\\xf0\\x9f\\xa4\\xad','').replace('\\xf0\\x9f\\x98\\xb5','').replace('\\xe2\\x80\\x8d','').replace('\\xf0\\x9f\\x92\\xab','').replace('\\xf0\\x9f\\xa4\\xa6','').replace('\\xe2\\x80\\x8d','').replace('\\xe2\\x99\\x80','').replace('\\xef\\xb8\\x8f','').replace('\\xf0\\x9f\\x92\\x99','').replace('\\xe2\\x9d\\xa4','').replace('\\xef\\xb8\\x8f','').replace('\\xe2\\x9d\\xa4','').replace('\\xef\\xb8\\x8f','').replace('\\xf0\\x9f\\x8f\\xbb','').replace('\\xe2\\x9d\\xa4','').replace('\\xef\\xb8\\x8f','').replace('\\xe2\\xa4\\xb5','').replace('\\xef\\xb8\\x8f','').replace('\\xf0\\x9f\\x92\\x81','').replace('\\xf0\\x9f\\x8f\\xbb','').replace('\\xe2\\x80\\x8d','').replace('\\xe2\\x99\\x82','').replace('\\xef\\xb8\\x8f','').replace('\\xf0\\x9f\\x96\\x95','').replace('\\xe2\\x99\\xa6','').replace('\\xef\\xb8\\x8f','').replace('\\xe2\\x99\\xa6','').replace('\\xef\\xb8\\x8f','').replace('\\xf0\\x9f\\x8c\\xba','').replace('\\xf0\\x9f\\x98\\x8f','').replace('\\xf0\\x9f\\xa4\\xa6','').replace('\\xf0\\x9f\\x8f\\xbb','').replace('\\xe2\\x80\\x8d','').replace('\\xe2\\x99\\x80','').replace('\\xef\\xb8\\x8f','')
#     no_unicode_line = no_unicode_line.replace('\\xf0\\x9f\\x8f\\xbd','').replace('\\xf0\\x9f\\x87\\xae\\xf0\\x9f\\x87\\xb3','').replace('\\xf0\\x9f\\x91\\x86','').replace('\\xf0\\x9f\\x8c\\xb9\\xf0\\x9f\\xa5\\x82','').replace('\\xe3\\x80\\x80','').replace('\\xf0\\x9f\\xa4\\xb7','').replace('\\xe2\\x80\\x8d','').replace('\\xe2\\x99\\x80','').replace('\\xef\\xb8\\x8f','').replace('\\xe2\\x9d\\xa4','').replace('\\xef\\xb8\\x8f','').replace('\\xf0\\x9f\\x87\\xb1','').replace('\\xf0\\x9f\\x87\\xb7','').replace('\\xe2\\x9d\\xa4','').replace('\\xef\\xb8\\x8f','').replace('\\xf0\\x9f\\x99\\x81','').replace('\\xf0\\x9f\\xa4\\xa5','').replace('\\xf0\\x9f\\xa4\\xa5','').replace('\\xf0\\x9f\\x87\\xb7','').replace('\\xf0\\x9f\\x87\\xba','').replace('\\xf0\\x9f\\x87\\xb7','').replace('\\xf0\\x9f\\x87\\xba','').replace('\\xf0\\x9f\\x91\\xb9','').replace('\\xf0\\x9f\\xa4\\xad','').replace('\\xf0\\x9f\\xa4\\xad','').replace('\\xf0\\x9f\\x98\\xb5','').replace('\\xe2\\x80\\x8d','').replace('\\xf0\\x9f\\x92\\xab','').replace('\\xf0\\x9f\\xa4\\xa6','').replace('\\xe2\\x80\\x8d','').replace('\\xe2\\x99\\x80','').replace('\\xef\\xb8\\x8f','').replace('\\xf0\\x9f\\x92\\x99','').replace('\\xe2\\x9d\\xa4','').replace('\\xef\\xb8\\x8f','').replace('\\xe2\\x9d\\xa4','').replace('\\xef\\xb8\\x8f','').replace('\\xf0\\x9f\\x8f\\xbb','').replace('\\xe2\\x9d\\xa4','').replace('\\xef\\xb8\\x8f','').replace('\\xe2\\xa4\\xb5','').replace('\\xef\\xb8\\x8f','').replace('\\xf0\\x9f\\x92\\x81','').replace('\\xf0\\x9f\\x8f\\xbb','').replace('\\xe2\\x80\\x8d','').replace('\\xe2\\x99\\x82','').replace('\\xef\\xb8\\x8f','').replace('\\xf0\\x9f\\x96\\x95','').replace('\\xe2\\x99\\xa6\\xef\\xb8\\x8f','').replace('\\xe2\\x99\\xa6','').replace('\\xef\\xb8\\x8f','').replace('\\xf0\\x9f\\x8c\\xba','').replace('\\xf0\\x9f\\x98\\x8f','').replace('\\xf0\\x9f\\xa4\\xa6','').replace('\\xf0\\x9f\\x8f\\xbb','').replace('\\xe2\\x80\\x8d','').replace('\\xe2\\x99\\x80','').replace('\\xef\\xb8\\x8f','')
#     no_unicode_line = no_unicode_line.replace('\\xe2\\x80\\x9c','').replace('\\xe2\\x80\\x9d','').replace('\\xe2\\x80\\xa6','').replace('\\xe2\\x80\\x99',"'").replace('\\xf0\\x9f\\x9a\\xa8','').replace('\\xc2\\xa0','').replace('\\xe2\\x80\\x98','').replace('\\xe2\\x80\\x94','').replace('\\xf0\\x9f\\x98\\xa2','').replace('\\xf0\\x9f\\x98\\xa2','').replace('\\xf0\\x9f\\x91\\x87','').replace('\\xf0\\x9f\\x87\\xba','').replace('\\xf0\\x9f\\x87\\xb8','').replace('\\xf0\\x9f\\x8f\\x9b','').replace('\\xf0\\x9f\\x98\\x86','').replace('\\xf0\\x9f\\xa4\\xa3','').replace('\\xf0\\x9f\\xa7\\xb5','').replace('\\xf0\\x9f\\xa4\\x94','').replace('\\xf0\\x9f\\x98\\x82','').replace('\\xf0\\x9f\\x92\\xaf','').replace('\\xf0\\x9f\\x94\\xa5','').replace('\\xf0\\x9f\\x8e\\xa4','').replace('\\xe2\\xac\\x87','').replace('\\xef\\xb8\\x8f','').replace('\\xf0\\x9f\\x98\\xa1','').replace('\\xf0\\x9f\\x99\\x84','').replace('\\xc2\\xbb','').replace('\\xe2\\x98\\x84','').replace('\\xef\\xb8\\x8f','').replace('\\xf0\\x9f\\xa5\\xba','').replace('\\xf0\\x9f\\x8c\\x8a','').replace('\\xf0\\x9f\\x8c\\x8a','').replace('\\xe2\\x9c\\x8c','').replace('\\xf0\\x9f\\x8f\\xbf','').replace('\\xe2\\x9a\\xa1','').replace('\\xef\\xb8\\x8f','').replace('\\xf0\\x9f\\x98\\x82','').replace('\\xf0\\x9f\\xa7\\xb5','').replace('\\xf0\\x9f\\x91\\x8d','').replace('\\xf0\\x9f\\x8f\\xbb','').replace('\\xf0\\x9f\\x8f\\xbc','').replace('\\xf0\\x9f\\xa4\\xac','').replace('\\xf0\\x9f\\x98\\xb3','').replace('\\xf0\\x9f\\x98\\x80','').replace('\\xf0\\x9f\\x93\\x96','').replace('\\xf0\\x9f\\xa4\\xb7','').replace('\\xf0\\x9f\\x8f\\xbd','').replace('\\xe2\\x80\\x8d','').replace('\\xe2\\x99\\x80','').replace('\\xef\\xb8\\x8f','').replace('\\xf0\\x9f\\x99\\x8f','').replace('\\xf0\\x9f\\x8f\\xbb','').replace('\\xf0\\x9f\\xa4\\xa1','').replace('\\xf0\\x9f\\x99\\x8f','').replace('\\\\xf0\\x9f\\x98\\x8a','').replace('\\xf0\\x9f\\x98\\x82','').replace('\\xf0\\x9f\\x92\\xaf','').replace('\\xf0\\x9f\\x94\\xa5','').replace('\\xf0\\x9f\\x8e\\xa4','').replace('\\xe2\\xac\\x87','').replace('\\xef\\xb8\\x8f','').replace('\\xf0\\x9f\\x98\\xa1','').replace('\\xf0\\x9f\\x99\\x84','').replace('\\xc2\\xbb','').replace('\\xe2\\x98\\x84','').replace('\\xef\\xb8\\x8f','').replace('\\xf0\\x9f\\xa5\\xba','').replace('\\xf0\\x9f\\x8c\\x8a','').replace('\\xf0\\x9f\\x8c\\x8a','').replace('\\xf0\\x9f\\x8f\\xbb','').replace('\\xf0\\x9f\\x8f\\xbb','').replace('\\xf0\\x9f\\x94\\xa5','').replace('\\xf0\\x9f\\x98\\x85','').replace('\\xe2\\xac\\x87','').replace('\\xef\\xb8\\x8f','').replace('\\xf0\\x9f\\x92\\x95','').replace('\\xf0\\x9f\\x91\\x80','').replace('\\xe2\\x9c\\x85','').replace('\\xc2\\xb0','').replace('\\xf0\\x9f\\xa4\\xa6','').replace('\\xf0\\x9f\\x8f\\xbb','').replace('\\xe2\\x80\\x8d','').replace('\\xe2\\x99\\x80\\xef\\xb8\\x8f','').replace('\\xf0\\x9f\\x8c\\xb9','').replace('\\xe3\\x80\\x80','').replace('\\xe2\\x99\\xa5\\xef\\xb8\\x8f\\xe2\\x99\\xa5\\xef\\xb8\\x8f\\xe2\\x99\\xa5\\xef\\xb8\\x8f','').replace('\\xf0\\x9f\\x8c\\xb9\\xf0\\x9f\\xa5\\x82','').replace('\\xe3\\x80\\x80','').replace('\\xf0\\x9f\\xa4\\xb7\\xe2\\x80\\x8d\\xe2\\x99\\x80\\xef\\xb8\\x8f','').replace('\\xe2\\x9d\\xa4\\xef\\xb8\\x8f\\xf0\\x9f\\x87\\xb1\\xf0\\x9f\\x87\\xb7\\xe2\\x9d\\xa4\\xef\\xb8\\x8f','').replace('\\xf0\\x9f\\x99\\x81','').replace('\\xf0\\x9f\\xa4\\xa5\\xf0\\x9f\\xa4\\xa5\\xf0\\x9f\\x87\\xb7\\xf0\\x9f\\x87\\xba\\xf0\\x9f\\x87\\xb7\\xf0\\x9f\\x87\\xba\\xf0\\x9f\\x91\\xb9','').replace('\\xf0\\x9f\\xa4\\xad\\xf0\\x9f\\xa4\\xad\\xf0\\x9f\\x98\\xb5\\xe2\\x80\\x8d\\xf0\\x9f\\x92\\xab\\xf0\\x9f\\xa4\\xa6\\xe2\\x80\\x8d\\xe2\\x99\\x80\\xef\\xb8\\x8f','').replace('\\xf0\\x9f\\x92\\x99','').replace('\\xe2\\x9d\\xa4\\xef\\xb8\\x8f\\xe2\\x9d\\xa4\\xef\\xb8\\x8f','').replace('\\xf0\\x9f\\x8f\\xbb','').replace('\\xe2\\x9d\\xa4\\xef\\xb8\\x8f','').replace('\\xe2\\xa4\\xb5\\xef\\xb8\\x8f','').replace('\\xf0\\x9f\\x92\\x81','').replace('\\xf0\\x9f\\x8f\\xbb','').replace('\\xe2\\x80\\x8d','').replace('\\xe2\\x99\\x82','').replace('\\xef\\xb8\\x8f','').replace('\\xf0\\x9f\\x96\\x95','').replace('\\xe2\\x99\\xa6','').replace('\\xef\\xb8\\x8f','').replace('\\xe2\\x99\\xa6','').replace('\\xef\\xb8\\x8f','').replace('\\xf0\\x9f\\x8c\\xba','').replace('\\xf0\\x9f\\x98\\x8f','').replace('\\xf0\\x9f\\xa4\\xa6','').replace('\\xf0\\x9f\\x8f\\xbb','').replace('\\xe2\\x80\\x8d','').replace('\\xe2\\x99\\x80','').replace('\\xef\\xb8\\x8f','')
#     no_uniocde_line = no_unicode_line.replace('\\xf0\\x9f\\x98\\x8a','').replace('\\xf0\\x9f\\x8c\\x8d','').replace('\\xf0\\x9f\\xa9\\xb8','').replace('\\xf0\\x9f\\x92\\xb0','').replace('\\xf0\\x9f\\x98\\xb0','').replace('\\xf0\\x9f\\xa5\\x82','').replace('\\xf0\\x9f\\x92\\x81','').replace('\\xe2\\x80\\x8d','').replace('\\xe2\\x99\\x82','').replace('\\xef\\xb8\\x8f','').replace('\\xf0\\x9f\\x98\\xae','').replace('\\xf0\\x9f\\x92\\x80','').replace('\\xc2\\xad','').replace('\\xf0\\x9f\\x98\\xae','').replace('\\xf0\\x9f\\x92\\x80','').replace('\\xf0\\x9f\\x99\\x8c','').replace('\\xf0\\x9f\\x8d\\x84','').replace('\\xe2\\x80\\x93','')